### DISCUSSION TITLE
Number of output features in Classification problems/models.

### DISCUSSION THREAD

**Discussion by Heyymant** [mrdbourke/pytorch-deep-learning#782] [Q&A]
Why does the number of output features for a binary classification data set is one, whereas multiclass problem for e.g. : 3 has 3 output features?

**Comment by allen-ajith** [comment]
For binary classification we can set output features to 2, just as we set output features to the number of classes in a multiclass problem. 

[Here's why that is not preferable](https://stats.stackexchange.com/questions/207049/neural-network-for-binary-classification-use-1-or-2-output-neurons)

**Comment by Harshith-reddy-td** [comment]
In case of binary classification output is set to one which has a prob range of 0 to 100. we can consider if prob is 0 as one class and prob 100 as second. but that doesn't hold good for multiclass problem where you have to mention how many classes you have those many output features

**Comment by LuluW8071** [comment]
The output for binary classification will always be either 0 or 1.
While if you go further for `n` multiclass classification it should always be 0,1,2,3,...,(n-1) classes.

Example:
- Lets say in classification of dog vs cats its either dog(0) or cat(1) as a result classification needs not more than 1 class
- While in multiclass(>=3 classes) dog(0), cats(1) and mouse(2) now needs 3 classes
